src/main/java/org/opensearch/security/user/PluginUser.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.user;
+
+import java.io.IOException;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+
+public class PluginUser extends User {
```

Comment:
Is it necessary to sub-class `User` for this?

One thing on my to do list is to write an RFC to propose to make `User` objects immutable - because it helps with stability and with peformance (no need of synchronized blocks) for thread safety.

Immutable objects however are easier to implement when you just have a single class which you have full control over - and do not need to support potential subclasses.

Reply:
I could use a similar pattern with service accounts which [uses the attribute map](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/user/User.java#L290-L298), but how can we assure that a regular user won't spoof the attribute map?

Reply:
How about just adding a new `type` attribute or something like that?